### PR TITLE
simplify credentials manager

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

Simplify credentials manager logic

# How

- use only runActionAsync
- replace pushNextAction with similar logic in actions that need that
- CredentialsManager just runs `runAsync` (Left that class there to simplify mocking in tests)
- runActionAsync can return values

# Test Plan

run build and few actions in credentials manager

### CODE REVIEW

File: packages/eas-cli/src/credentials/manager/ManageIos.ts [hunk 1]
```diff
@@ -32,66 +33,76 @@ enum ActionType {
...
+            },
+            {
+              value: ActionType.RemoveDistributionCertificate,
+              title: 'Remove Distribution Certificate',
+            },
+            {
+              value: ActionType.UpdateDistributionCertificate,
+              title: 'Update Distribution Certificate',
+            },
+          ],
```
Comment (dsokal) [inline packages/eas-cli/src/credentials/manager/ManageIos.ts:92]: Maybe an exit action would be useful? I assume the only way of exiting the manager is by pressing `CTRL + C`?

**Review by dsokal — APPROVED** [review]
👏

File: packages/eas-cli/src/commands/credentials.ts [hunk 2]
```diff
@@ -9,6 +9,6 @@ export default class Credentials extends Command {
 
   async run() {
     const ctx = await createCredentialsContextAsync(process.cwd(), {});
-    await runStandaloneCredentialsManagerAsync(ctx, new SelectPlatform());
+    new CredentialsManager(ctx).runActionAsync(new SelectPlatform());
```
Comment (quinlanj) [inline packages/eas-cli/src/commands/credentials.ts:12]:
```suggestion
    await (new SelectPlatform()).runAsync(ctx);
```
would this work as well? or were you planning to do this in a separate pr when you remove the credentials manager class?

**Review by quinlanj — APPROVED** [review]
🚀

File: packages/eas-cli/src/commands/credentials.ts (hunk 2 above)
Comment (wkozyra95) [reply packages/eas-cli/src/commands/credentials.ts:12]: yes it would work, but mocking in tests would require a lot of changes. Currently, I can just mock one function and check what type was passed to runActionAsync, but with this approach I need to mock all Actions that potentially can be triggered from the action I'm currently testing
